src/hooks/useFlags.ts

```diff
@@ -65,11 +65,16 @@ export type Args = {
 export function useArgs(args: string[], arg0: string): [Args, Flags & ConvenienceFlags] {
   if (flags) throw new Error("contract-violated")
 
-  if (/(.+\/|^)tea_(.+)$/.test(arg0)) {
-    args = [...args]  // args is usually the immutable `Deno.args`
-    //TODO apply muggle mode
-    const match = arg0.match(/tea_(.+)$/)!
-    args.unshift("-X", match[1])
+  const arg0_basename = new Path(arg0).basename()
+  if (arg0_basename != arg0) {
```

Comment:
Fixed the whitespace (and set my editor to use spaces instead of tabs).

I added a symlink test, which should solve the deno interpretation issue.

It kept going in an infinite loop, which didn't make any sense to me until I realized that it was adding the cwd to the PATH. When I cd'ed out of the directory that contained tea and the symlink it finally worked. So this has the potential to break unless we remove the cwd from the PATH. I'm not sure what to do about that.

Reply:
"I'm not sure what to do about that" means this is a design decision and I don't make those types of decisions for the project, not when I barely know the code. I was kind of [hoping for a -Y argument](https://github.com/teaxyz/cli/issues/170#issuecomment-1320352455) that had a much smaller search path. If the symlinks used that, it would solve this.

Reply:
This is the 2nd time I've worked collaboratively with GitHub and I don't have much experience with the workflow stuff. Am I supposed to click "Resolve conversation" to move it past the "Changes requested" stage?

There is one issue but I think it's independent of this PR so I opened a new issue https://github.com/teaxyz/cli/issues/199.